### PR TITLE
Improve Infinispan initialization

### LINKED ISSUE #24860: Initialize Infinispan earlier in the build chain
**Start the default ISPN cache earlier in the startup flow** 
We don't really need to rely on Keycloak session factory initialization, as only the new thread with the default ISPN cache is started. It doesn't work with any providers implementation, only the Microprofile config is required, and we don't need any "runtime" config properties. As we don't use native builds, we can simply move the execution of the recorded bytecode for cacheManager to the STATIC_INIT and change the order of these executions. 

Old cache startup:
old-ispn

New cache startup:
new_ispn

We can save around **~800ms for the startup.**

### PR DESCRIPTION
**Pull request by mabartos**
Closes https://github.com/keycloak/keycloak/issues/24860

**Start the default ISPN cache earlier in the startup flow** 
We don't really need to rely on Keycloak session factory initialization, as only the new thread with the default ISPN cache is started. It doesn't work with any providers implementation, only the Microprofile config is required, and we don't need any "runtime" config properties. As we don't use native builds, we can simply move the execution of the recorded bytecode for cacheManager to the STATIC_INIT and change the order of these executions. 

Old cache startup:
old-ispn

New cache startup:
new_ispn

We can save around **~800ms for the startup.**

I've tried to do some experiments with the cache:
1) **One instance with the `join_timeout=10`**. The instance becomes the coordinator of the cluster almost immediately and doesn't affect any other startup items executions - works properly
2) **Cluster of 5 Keycloak instances.** The other instances are properly connected, and there's no issue.

@pedroigor @vmuzikar Could you please check it? Thanks!

### CODE REVIEW

**Comment by vmuzikar** [comment]
Per offline discussion, let's hold this off until after 22 is released.

**Review by pedroigor — COMMENTED** [review]
@mabartos I have also tried this in the past without great improvements. I think we started as STATIC_INIT and then changed to RUNTIME_INIT. Not 100% sure.

But if now we have a so great improvement, +1for having it.

**Comment by abstractj** [comment]
@vmuzikar should we remove the hold label now?

File: quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/CacheBuildSteps.java
Comment (mabartos) [inline quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/CacheBuildSteps.java:67]: A follow-up task will be the possibility to parse the file in the build time and not just read it. It'll save another ~100ms.

Comment (ahus1) [reply quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/CacheBuildSteps.java:67]: While the default configuration looks a build time configuration, it usually contains placeholders which are supposed to change at runtime. Please double-check with the store team if this is the way to go to avoid going into a direction which wouldn't work for Infinispan.

**Comment by mabartos** [comment]
### Startup time improvements

In this startup time analysis, we can consider two testing samples: **Old** and **New**. 

Reference: 
**Old**  - distribution without these changes; `upstream/main`. 
**New** - distribution with these changes. 

I've created a docker image and 9x executed it in order to get the results.

#### Results
  
  
 You can notice the startup time in seconds in the graph, where these values also depends on the current utilization of CPU by other tasks. I tried to decrease the number of other tasks, so the AVG CPU utilization was ~1%. 
  
  <table>
    <tr>
       <td align="center">Round</td>
       <td align="center">1</td>
       <td align="center">2</td>
       <td align="center">3</td>
       <td align="center">4</td>
       <td align="center">5</td>
       <td align="center">6</td>
       <td align="center">7</td>
       <td align="center">8</td>
       <td align="center">9</td> 
       <td align="center">Average</td>             
    </tr>
    <tr>
      <td>Old (s)</td>
      <td>4.873</td>
      <td>4.903</td>
      <td>5.188</td>
      <td>4.732</td>
      <td>4.751</td>
      <td>4.78</td>
      <td>4.911</td>
      <td>4.846</td>
      <td>5.049</td>
      <td><b>4.892</b></td>
    </tr>
    <tr>
      <td>New (s)</td>
      <td>3.935</td>
      <td>3.925</td>
      <td>4.088</td>
      <td>4.202</td>
      <td>3.981</td>
      <td>3.962</td>
      <td>3.983</td>
      <td>4.537</td>
      <td>3.959</td>
      <td><b>4.063</b></td>
    </tr>
  </table>
  
  The average overall startup time improvement is **-829ms**, which is basically the expected one. 
  
 ---------------------------
I executed 5 pods in the K8s cluster and haven't spotted any issues. 

@keycloak/cloud-native-maintainers @pedroigor Could you please review it? Thanks

**Comment by mabartos** [comment]
Could someone from the @keycloak/store review it, please?

**Comment by mabartos** [comment]
@keycloak/store @keycloak/cloud-native Are you ok with these changes?

**Review by mhajas — APPROVED** [review]
Thank you @mabartos. This change makes sense to me! I think we should have another review from the @keycloak/cloud-native team but from store team I don't see any problem with this.

**Comment by Pepo48** [comment]
LGTM, thanks @mabartos! Looking at the numbers again it's a nice improvement, indeed.

**Review by Pepo48 — APPROVED** [review]

**Pull request merged**
